### PR TITLE
ME-1602: add client methods for terraform socket resource

### PR DESCRIPTION
**Pull request by waltzofpearls**
- add client methods for terraform `border0_socket` resource
- fix a bug where listener failed to detected and gracefully handle the
  error when socket didn't exist, now with the fix, it should be able to
  automatically create the socket if not exists
- http client now try to catch fallback error `message` if
  `error_message` field is absent or empty
- examples are now ordered and prefixed with numbers
- add more examples for api client: delete socket, list all sockets
- add more examples for `border0.Listen`: stdlib http, gin, echo and
  fiber

### CODE REVIEW

File: client/http_client.go [hunk 1]
```diff
@@ -103,16 +103,33 @@ func APIErrorFrom(resp *http.Response) Error {
...
 	return apiErr
 }
 
 // Error is an error returned by the API server.
 type Error struct {
-	Code    int    `json:"status_code"`
-	Message string `json:"error_message"`
+	Code     int    `json:"status_code"`
+	Message  string `json:"error_message"`
+	Fallback string `json:"message"`
```
Comment (adrianosela) [inline client/http_client.go:121]: When does this happen? Seems like something we should remove in the api side if its inconsistent?

File: client/socket.go [hunk 2]
```diff
@@ -50,9 +89,23 @@ func (api *APIClient) SignSocketKey(ctx context.Context, idOrName string, in *So
...
 type Socket struct {
-	Name       string `json:"name"`
-	SocketID   string `json:"socket_id"`
-	SocketType string `json:"socket_type"`
+	Name                 string            `json:"name"`
+	SocketID             string            `json:"socket_id"`
+	SocketType           string            `json:"socket_type"`
+	Description          string            `json:"description,omitempty"`
+	UpstreamType         string            `json:"upstream_type,omitempty"`
+	UpstreamHTTPHostname string            `json:"upstream_http_hostname,omitempty"`
```
Comment (adrianosela) [inline client/socket.go:97]: This seems inconsistent, can we remove it?

**Review by adrianosela — APPROVED** [review]

File: client/http_client.go (hunk 1 above)
Comment (waltzofpearls) [reply client/http_client.go:121]: Run the following command from api repo's directory, and you will see those.

```shell
grep -irn --color '"message"' .
```

File: client/socket.go (hunk 2 above)
Comment (waltzofpearls) [reply client/socket.go:97]: This is for HTTP socket, for example

```shell
border0 socket create \
  --type http \
  --upstream_type https \
  --upstream_http_hostname www.bbc.com \
  --name bbc 
```

I can change that later once we made changes in the api.

**Pull request merged**